code/go/internal/validator/semantic/validate_agent_version_required.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package semantic
+
+import (
+	"fmt"
+
+	"github.com/elastic/package-spec/v3/code/go/internal/fspath"
+	"github.com/elastic/package-spec/v3/code/go/internal/pkgpath"
+	"github.com/elastic/package-spec/v3/code/go/pkg/specerrors"
+)
+
+var (
+	errAgentVersionConditionMissing = fmt.Errorf("agent.version condition is required")
+	errAgentVersionIncorrectType    = fmt.Errorf("manifest agent version is not a string")
+)
+
+// ValidateMinimumAgentVersion checks that the package manifest includes the agent.version condition.
+// This is required for integration packages since version 3.6.0 of the spec.
+func ValidateMinimumAgentVersion(fsys fspath.FS) specerrors.ValidationErrors {
+	manifest, err := readManifest(fsys)
+	if err != nil {
+		return specerrors.ValidationErrors{specerrors.NewStructuredError(err, specerrors.UnassignedCode)}
+	}
+
+	agentVersionCondition, err := getAgentVersionCondition(*manifest)
+	if err != nil {
+		return specerrors.ValidationErrors{specerrors.NewStructuredError(err, specerrors.UnassignedCode)}
+	}
+
+	if agentVersionCondition == "" {
+		return specerrors.ValidationErrors{specerrors.NewStructuredError(errAgentVersionConditionMissing, specerrors.UnassignedCode)}
+	}
+
+	return nil
+}
+
+func getAgentVersionCondition(manifest pkgpath.File) (string, error) {
+	val, err := manifest.Values("$.conditions[\"agent.version\"]")
+	if err != nil {
+		val, err = manifest.Values("$.conditions.agent.version")
+		if err != nil {
+			return "", nil
+		}
+	}
```

Comment:
We are not returning these errors, do they happen when the field is not found? Should we handle other cases? Though I see we do the same in other places.

Reply:
Yeah, I would bring the check for requirement to the spec, and leave here only the validation of the format.